src/org/spdx/html/LicenseTOCJSONFile.java

```diff
@@ -0,0 +1,150 @@
+		private String reference;
+		private String refNumber;
+		private String licenseId;
+		private boolean osiApproved;
+		private String licenseName;
+		
+		public ListedSpdxLicense() {
+			reference = null;
+			refNumber = null;
+			licenseId = null;
+			osiApproved = false;
+			licenseName = null;
+		}
+		
+		public ListedSpdxLicense(String reference, String refNumber, 
+				String licenseId, boolean isOsiApproved, String licenseName) {
+			this.reference = reference;
+			this.refNumber = refNumber;
+			this.licenseId = licenseId;
+			this.osiApproved = osiApproved;
+			this.licenseName = licenseName;
+		}
+
+		public String getReference() {
+			return reference;
+		}
+
+		public void setReference(String reference) {
+			this.reference = reference;
+		}
+
+		public String getRefNumber() {
+			return refNumber;
+		}
+
+		public void setRefNumber(String refNumber) {
+			this.refNumber = refNumber;
+		}
+
+		public String getLicenseId() {
+			return licenseId;
+		}
+
+		public void setLicenseId(String licenseId) {
+			this.licenseId = licenseId;
+		}
+
+		public boolean getOsiApproved() {
+			return osiApproved;
+		}
+
+		public void setOsiApproved(boolean osiApproved) {
+			this.osiApproved = osiApproved;
+		}
+
+		public String getLicenseName() {
+			return licenseName;
+		}
+
+		public void setLicenseName(String licenseName) {
+			this.licenseName = licenseName;
+		}
+	}
+	
+	ArrayList<ListedSpdxLicense> listedLicenses = new ArrayList<ListedSpdxLicense>();
+	
+	private int currentRefNumber = 1;
+
+	String version;
+	String releaseDate;
+
+	public LicenseTOCJSONFile(String version, String releaseDate) {
+		this.version = version;
+		this.releaseDate = releaseDate;
+	}
+
+	public void addLicense(SpdxListedLicense license, String licHTMLReference) {
+		listedLicenses.add(new ListedSpdxLicense(licHTMLReference, String.valueOf(this.currentRefNumber), 
+				license.getLicenseId(), license.isOsiApproved(), license.getName()));
+		currentRefNumber++;
+	}
+
+	public void writeToFile(File jsonFile) throws IOException {
+		FileWriter writer = null;
+		if (!jsonFile.exists()) {
+			if (!jsonFile.createNewFile()) {
+				throw(new IOException("Can not create new file "+jsonFile.getName()));
+			}
+		}
+		try {
+			JSONObject jsonObject = new JSONObject();
+			jsonObject.put("version", version);
+			jsonObject.put("releaseDate", releaseDate);
+			JSONArray licensesList = new JSONArray();
+			for (ListedSpdxLicense license : listedLicenses) {
+				JSONObject licenseJSON = new JSONObject();
+				licenseJSON.put("reference", license.getReference());
+				licenseJSON.put("referenceNumber", license.getRefNumber());
+				licenseJSON.put("id", license.getLicenseId());
+				licenseJSON.put("osiApproved", license.getOsiApproved());
+				licenseJSON.put("name", license.getLicenseName());
+				licensesList.add(licenseJSON);
+			}
+			jsonObject.put("licenses", licensesList);
+			writer = new FileWriter(jsonFile);
+			writer.write(jsonObject.toJSONString());

```

Comment:
>  But I for one would like to move forward on the list without waiting for full texts, in part because rubygems/rubygems#1249 is waiting.

:+1: 

> Perhaps you'd like open an issue on spdx/tools?

It would be nice to have those easily accessible from the official source, but I don't have time to deal with the back and forth of such an issue and the current code I use to get it works fine for me.